OWML.ModHelper/ModMergedConfig.cs

```diff
@@ -60,120 +57,83 @@ public void SaveToStorage()
-            var userValue = _userConfig.GetSettingsValue<object>(key);
-            if (userValue is JValue userJValue)
+            if (!_defaultConfig.Settings.ContainsKey(key))
             {
-                userValue = userJValue.Value;
+                return false;
             }
-            _userConfig.Settings[key] = modSetting;
-
-            if (IsNumber(userSetting) && IsNumber(modSetting))
+            var userValue = _userConfig.Settings[key];
+            var defaultValue = _defaultConfig.Settings[key];
+            var defaultInnerValue = GetInnerValue(defaultValue);
+            if (IsNumeric(userValue) && IsNumeric(defaultInnerValue))
             {
-                _userConfig.SetSettingsValue(key, Convert.ToDouble(userValue));
                 return true;
             }
-
-            if (IsBoolean(userSetting) && IsBoolean(modSetting))
-            {
-                _userConfig.SetSettingsValue(key, Convert.ToBoolean(userValue));
-                return true;
-            }
-            return false;
+            return userValue.GetType() == defaultInnerValue.GetType();
         }
 
-        private bool IsNumber(object setting)
+        private void MakeConfigConsistentWithDefault()
         {
-            return setting is JObject settingObject
-                ? settingObject["type"].ToString() == "slider"
-                : new[] { typeof(long), typeof(int), typeof(float), typeof(double) }.Contains(setting.GetType());
+            _userConfig.Settings.Keys.ToList().ForEach(key =>
+            {
+                if (!IsSettingConsistentWithDefault(key))
+                {
+                    _userConfig.Settings.Remove(key);
+                }
+            });
         }
 
-        private bool IsBoolean(object setting)
+        private void FixConfigs()
         {
-            return setting is JObject settingObject
-                ? settingObject["type"].ToString() == "toggle"
-                : setting is bool;
+            MakeConfigConsistentWithDefault();
+            SaveToStorage();
         }
 
-        private bool IsSettingSameType(object settingValue1, object settingValue2)
+        private object GetConvertedSelectorValue(string key, object value)
         {
-            return settingValue1.GetType() == settingValue2.GetType() &&
-                   (!(settingValue1 is JObject obj1) || !(settingValue2 is JObject obj2) ||
-                    (string)obj1["type"] == (string)obj2["type"]);
+            if (!_defaultConfig.Settings.ContainsKey(key))
+            {
+                return null;
+            }
+            var defaultValue = _defaultConfig.Settings[key];
+            if (defaultValue is JObject defaultObjectValue && defaultObjectValue["type"].ToString() == "selector")
+            {
+                var innerValue = defaultObjectValue["value"];
+                return innerValue.Type == JTokenType.Integer ? int.Parse(value.ToString()) : value;
+            }
+            return null;
```

Comment:
This is the hackiest part of this change. It was already broken before, but in a way that didn't affect anything. Our selectors are created as extending `IModInput<string>`, which doesn't make much sense since selectors can have string or int values. I couldn't find a good way to fix this at the source, like making our selectors generic, or having two types of selector. So instead I'm parsing the string value back to int for number selectors. The type of selector is determined by the type of the default value.